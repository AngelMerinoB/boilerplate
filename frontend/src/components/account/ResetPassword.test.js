import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { ResetPassword } from './ResetPassword';
Enzyme.configure({ adapter: new Adapter() });

const noop = () => {};

describe('<ResetPassword />', () => {
  test('should render <ResetPassword /> without errors', () => {
    const wrapper = Enzyme.shallow(
      <ResetPassword location={{ search: '' }} resetPassword={noop} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the reset message', () => {
    const wrapper = Enzyme.shallow(
      <ResetPassword location={{ search: '?reset' }} resetPassword={noop} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
