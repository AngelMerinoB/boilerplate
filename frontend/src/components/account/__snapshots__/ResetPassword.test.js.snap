// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> should render <ResetPassword /> without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetPassword
    location={
        Object {
            "search": "",
          }
    }
    resetPassword={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Form
        ErrorList={[Function]}
        liveValidate={false}
        noHtml5Validate={true}
        noValidate={false}
        onSubmit={[Function]}
        safeRenderCompletion={false}
        schema={
                Object {
                        "properties": Object {
                          "password": Object {
                            "title": "New password",
                            "type": "string",
                          },
                        },
                        "required": Array [
                          "password",
                        ],
                        "title": "Password reset",
                        "type": "object",
                      }
        }
        uiSchema={
                Object {
                        "password": Object {
                          "ui:widget": "password",
                        },
                      }
        }
>
        <fieldset>
                <button
                        className="btn btn-primary btn-lg btn-block"
                        type="submit"
                >
                        Change your password
                </button>
        </fieldset>
</Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ErrorList": [Function],
        "children": <fieldset>
          <button
                    className="btn btn-primary btn-lg btn-block"
                    type="submit"
          >
                    Change your password
          </button>
</fieldset>,
        "liveValidate": false,
        "noHtml5Validate": true,
        "noValidate": false,
        "onSubmit": [Function],
        "safeRenderCompletion": false,
        "schema": Object {
          "properties": Object {
            "password": Object {
              "title": "New password",
              "type": "string",
            },
          },
          "required": Array [
            "password",
          ],
          "title": "Password reset",
          "type": "object",
        },
        "uiSchema": Object {
          "password": Object {
            "ui:widget": "password",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-primary btn-lg btn-block"
            type="submit"
>
            Change your password
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Change your password",
            "className": "btn btn-primary btn-lg btn-block",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Change your password",
          "type": "button",
        },
        "type": "fieldset",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Form
          ErrorList={[Function]}
          liveValidate={false}
          noHtml5Validate={true}
          noValidate={false}
          onSubmit={[Function]}
          safeRenderCompletion={false}
          schema={
                    Object {
                              "properties": Object {
                                "password": Object {
                                  "title": "New password",
                                  "type": "string",
                                },
                              },
                              "required": Array [
                                "password",
                              ],
                              "title": "Password reset",
                              "type": "object",
                            }
          }
          uiSchema={
                    Object {
                              "password": Object {
                                "ui:widget": "password",
                              },
                            }
          }
>
          <fieldset>
                    <button
                              className="btn btn-primary btn-lg btn-block"
                              type="submit"
                    >
                              Change your password
                    </button>
          </fieldset>
</Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ErrorList": [Function],
          "children": <fieldset>
            <button
                        className="btn btn-primary btn-lg btn-block"
                        type="submit"
            >
                        Change your password
            </button>
</fieldset>,
          "liveValidate": false,
          "noHtml5Validate": true,
          "noValidate": false,
          "onSubmit": [Function],
          "safeRenderCompletion": false,
          "schema": Object {
            "properties": Object {
              "password": Object {
                "title": "New password",
                "type": "string",
              },
            },
            "required": Array [
              "password",
            ],
            "title": "Password reset",
            "type": "object",
          },
          "uiSchema": Object {
            "password": Object {
              "ui:widget": "password",
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary btn-lg btn-block"
              type="submit"
>
              Change your password
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Change your password",
              "className": "btn btn-primary btn-lg btn-block",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Change your password",
            "type": "button",
          },
          "type": "fieldset",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ResetPassword /> should render the reset message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetPassword
    location={
        Object {
            "search": "?reset",
          }
    }
    resetPassword={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card shadow-lg p-3 mb-5 bg-white rounded"
        style={
                Object {
                        "margin": "100px auto",
                        "width": "20rem",
                      }
        }
>
        <div
                className="card-body"
        >
                <h5
                        className="card-title"
                >
                        <i
                                className="fas fa-check"
                        />
                </h5>
                <p
                        className="card-text"
                >
                        Your password has been successfully changed.
                </p>
                <Link
                        className="btn btn-primary"
                        replace={false}
                        to="/account/login"
                >
                        Log in with your new password
                </Link>
        </div>
</div>,
      "className": "container text-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-body"
>
          <h5
                    className="card-title"
          >
                    <i
                              className="fas fa-check"
                    />
          </h5>
          <p
                    className="card-text"
          >
                    Your password has been successfully changed.
          </p>
          <Link
                    className="btn btn-primary"
                    replace={false}
                    to="/account/login"
          >
                    Log in with your new password
          </Link>
</div>,
        "className": "card shadow-lg p-3 mb-5 bg-white rounded",
        "style": Object {
          "margin": "100px auto",
          "width": "20rem",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="card-title"
>
              <i
                            className="fas fa-check"
              />
</h5>,
            <p
              className="card-text"
>
              Your password has been successfully changed.
</p>,
            <Link
              className="btn btn-primary"
              replace={false}
              to="/account/login"
>
              Log in with your new password
</Link>,
          ],
          "className": "card-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-check"
/>,
              "className": "card-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-check",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your password has been successfully changed.",
              "className": "card-text",
            },
            "ref": null,
            "rendered": "Your password has been successfully changed.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Log in with your new password",
              "className": "btn btn-primary",
              "replace": false,
              "to": "/account/login",
            },
            "ref": null,
            "rendered": "Log in with your new password",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card shadow-lg p-3 mb-5 bg-white rounded"
          style={
                    Object {
                              "margin": "100px auto",
                              "width": "20rem",
                            }
          }
>
          <div
                    className="card-body"
          >
                    <h5
                              className="card-title"
                    >
                              <i
                                        className="fas fa-check"
                              />
                    </h5>
                    <p
                              className="card-text"
                    >
                              Your password has been successfully changed.
                    </p>
                    <Link
                              className="btn btn-primary"
                              replace={false}
                              to="/account/login"
                    >
                              Log in with your new password
                    </Link>
          </div>
</div>,
        "className": "container text-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card-body"
>
            <h5
                        className="card-title"
            >
                        <i
                                    className="fas fa-check"
                        />
            </h5>
            <p
                        className="card-text"
            >
                        Your password has been successfully changed.
            </p>
            <Link
                        className="btn btn-primary"
                        replace={false}
                        to="/account/login"
            >
                        Log in with your new password
            </Link>
</div>,
          "className": "card shadow-lg p-3 mb-5 bg-white rounded",
          "style": Object {
            "margin": "100px auto",
            "width": "20rem",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-title"
>
                <i
                                className="fas fa-check"
                />
</h5>,
              <p
                className="card-text"
>
                Your password has been successfully changed.
</p>,
              <Link
                className="btn btn-primary"
                replace={false}
                to="/account/login"
>
                Log in with your new password
</Link>,
            ],
            "className": "card-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-check"
/>,
                "className": "card-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-check",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your password has been successfully changed.",
                "className": "card-text",
              },
              "ref": null,
              "rendered": "Your password has been successfully changed.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log in with your new password",
                "className": "btn btn-primary",
                "replace": false,
                "to": "/account/login",
              },
              "ref": null,
              "rendered": "Log in with your new password",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
